### PR TITLE
Patch polymorphic / uncertain state type

### LINKED ISSUE #171: get_characters() fails with an error in dplyr::bind_rows()
The [following line](https://github.com/ropensci/RNeXML/blob/4341d2465a4bbaf1693c5bc23209440a44a2c807/R/get_characters.R#L59) in `get_characters.R` fails with `Column `symbol` can't be converted from integer to character`:

```R
states <- dplyr::bind_rows(states, polymorph)
Error in bind_rows_(x, .id) : 
  Column `symbol` can't be converted from integer to character
```

This seems to happen only in R 3.5.+, not in R 3.3.x. I don't know whether this happens for all inputs with polymorphic characters or only some. I came across this from the Rphenoscape vignette, where it breaks rendering the vignette:

```R
install_github("xu-hong/rphenoscape")
library(rphenoscape)
nex <- pk_get_ontotrace_xml(taxon = c("Ictalurus", "Ameiurus"), entity = "fin spine")
m <- pk_get_ontotrace(nex)
```

The last instruction internally calls `get_characters()`:
```R
m <- get_characters(nex, rownames_as_col = TRUE,
                    otu_id = TRUE, otus_id = TRUE)
```

Which then results in the given error.

### PR DESCRIPTION
**Pull request by cboettig**
@hlapp I think this should now fix #171 / #172, but please verify.  As mentioned in the commit note, this  is just a really small fix to a bug in the S4 class definition of `state`, which defined `symbol` as integer type for standard states (e.g. `symbol="1"`), but as character type for polymorphic states (e.g.  `symbol="1 or 0"`).  

Actually I believe this may actually be from what feels like an inconsistency in the NeXML schema.  
In a standard state it appears to me that `symbol` attribute on a state is technically typed as a [StandardToken](http://www.nexml.org/doc/schema-1/characters/standard/#StandardToken), which looks to me to be defined as a restriction of `xs:integer`, which is I think why I typed the symbol as integer in the first place. 

Meanwhile, in the NeXML schema defines `symbol` to be a `xs:string` in the case of a polymorphic state set, at least according to: http://www.nexml.org/doc/schema-1/characters/standard/#StandardUncertainStateSet

To me, this seems inconsistent, and is also the immediate cause of the bugs you identified, because R refuses to bind the integer symbol column to a character symbol column.  Declaring symbol to always be a character string as I've done in the PR resolves the issue, but technically appears to be a deviation from the types declared by the schema itself.  

cc'ing @rvosa for additional insight on this.

### CODE REVIEW

**Comment by hlapp** [comment]
Couple comments:

* That standard state symbols are typed as integers in NeXML seems appropriate given standard practice, though you might call it needlessly strict. That is, common exchange formats won't make it distinguishable whether the symbols in an integer-coded character column are type integer or string, and I doubt any R package for comparative analysis expects the states to be typed as integers or character vector so long as they can be turned into a factor. So I suspect there's no harm from typing these generally as character vectors as I suppose you do here. However, on input, symbols for standard states _will_ be integers in all practical cases, even if the schema didn't enforce that.
* My approach to fixing the type mismatch problem was to simply add
  ```R
  states <- states %>% dplyr::mutate_at(c("symbol"), as.character)
  ```
  and to do so only if there are polymorphic or uncertain states. This fixes the type problem too but without impact if there are no polymorphic and uncertain states. You don't think that's a good solution?
* This really only fixes the exception that's thrown, i.e., the issue reported in #171. It doesn't fix the other issues.

File: R/get_characters.R [hunk 1]
```diff
@@ -96,7 +96,7 @@ get_characters <- function(nex, rownames_as_col=FALSE, otu_id = FALSE, otus_id =
     dplyr::distinct() %>%
     dplyr::mutate_(.dots = setNames(list(~cellclass(xsi.type)), "class"))
   
-  for(i in dim(type)[1])
+  for(i in seq_along(seq(1,dim(type)[1])))
```
Comment (hlapp) [inline R/get_characters.R]: Wouldn't seq(length.out=dim(type)[1]) be simpler (and clearer)?

File: tests/testthat/test_comp_analysis.R [hunk 2]
```diff
@@ -7,7 +7,7 @@ library(geiger)
 test_that("We can extract tree and trait data to run fitContinuous and fitDiscrete", {
   nexml <- read.nexml(system.file("examples", "comp_analysis.xml", package="RNeXML"))
   traits <- get_characters(nexml)
-  tree <- get_trees(nexml)
+  tree <- get_trees(nexml)
```
Comment (hlapp) [inline tests/testthat/test_comp_analysis.R]: That's an unrelated whitespace-only change, right?

File: R/get_characters.R (hunk 1 above)
Comment (cboettig) [reply R/get_characters.R]: :+1:

File: tests/testthat/test_comp_analysis.R (hunk 2 above)
Comment (cboettig) [reply tests/testthat/test_comp_analysis.R]: whoops, yes,I  shouldn't be introducing random whitespace changes

**Comment by cboettig** [comment]
@hlapp Thanks.  I still don't follow the justification for the standard to type `symbol` attribute differently in polymorphic and non-polymorphic cases, that seems to be asking for trouble.  But given that is what the standard says, yes, your solution is probably better.  (I changed this first, assuming I had not followed the standard somehow, and only noticed later the standard really did type `symbol` as an integer.  

Would you want to send a PR introducing 

```r
states <- states %>% dplyr::mutate_at(c("symbol"), as.character)
```

?


but this does resolve #171 , yes? (`ERRROR Column `symbol` can't be converted from integer to character`).  Just not all the issues in #172.  

Yeah, I think #172 discusses this issue and also separate issues, might be easier to deal with the separate issues in separate issues.

**Comment by hlapp** [comment]
Yes I can do that.

**Comment by hlapp** [comment]
See #174.

**Comment by cboettig** [comment]
Closing in favor of #174
